java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -101,11 +102,39 @@ public ColumnView(DType type, long rows, Optional<Long> nullCount,
         || !nullCount.isPresent();
   }
 
+  /**
+   * Create a new column view based off of data already on the device. Ref count on the buffers
+   * is not incremented and none of the underlying buffers are owned by this view. The returned
+   * ColumnView is only valid as long as the underlying buffers remain valid. If the buffers are
+   * closed before this ColumnView is closed, it will result in undefined behavior.
+   *
+   * If ownership is needed, call {@link ColumnView#copyToColumnVector}
+   *
+   * @param type           the type of the vector
+   * @param rows           the number of rows in this vector.
+   * @param nullCount      the number of nulls in the dataset.
+   * @param dataBuffer     a host buffer required for nested types including strings and string
+   *                       categories. The ownership doesn't change on this buffer
+   * @param validityBuffer an optional validity buffer. Must be provided if nullCount != 0.
+   *                       The ownership doesn't change on this buffer
+   * @param offsetBuffer   The offsetbuffer for columns that need an offset buffer
+   */
+  public ColumnView(DType type, long rows, Optional<Long> nullCount,
```

Comment:
This API really confuses me, because why would we need to create a ColumnView that has offsets in it, but no child?  In List columns there is always a child.  We hacked it up badly for Strings in the java side by putting them together, but we should not have done that.

Reply:
This API mirrors ColumnVector's API. Strings column is a hybrid column on the Java side as we make it seem like it doesn't have children but has an offset buffer and a data buffer where on the cudf side it the parent only has a validity buffer, the data and offsets are children.

Ideally we should create a String column like we create a List, then we won't need a special case just for Strings, right now String is stuck in between the `nested` and `non-nested` type in Java. 

I don't know how else to do this other than changing how we are creating Strings

Reply:
Sorry I am not asking you to change how we do Strings. I am just trying to understand why these changes are being made, and why one API is deprecated in favor of this one. I see now that it follows the public API in ColumnVector and is a superset of the old API. I am still not convinced we need to deprecate the old API, but this is fine.